Xamarin.Forms.Platform.iOS/CollectionView/SelectableItemsViewController.cs

```diff
@@ -89,7 +115,40 @@ internal void UpdateSelectionMode()
 					break;
 			}
 
-			UpdateFormsSelection();
+			UpdateNativeSelection();
+		}
+
+		void SynchronizeNativeSelectionWithSelectedItems()
+		{
+			var selectedItems = SelectableItemsView.SelectedItems;
+			var selectedIndexPaths = CollectionView.GetIndexPathsForSelectedItems();
+
+			foreach (var path in selectedIndexPaths)
+			{
+				var itemAtPath = GetItemAtIndex(path);
+				if (ShouldNotBeSelected(itemAtPath, selectedItems))
+				{
+					CollectionView.DeselectItem(path, true);
+				}
+			}
+
+			foreach (var item in selectedItems)
+			{
+				SelectItem(item);
+			}
+		}
+
+		bool ShouldNotBeSelected(object item, IList<object> selectedItems)
+		{
+			for (int n = 0; n < selectedItems.Count; n++)
+			{
+				if (selectedItems[n] == item)
+				{
+					return false;
+				}
+			}
+
+			return true;
```

Comment:
can simply call the `!selectedItems.Contais(item)` ?

Reply:
we don't want to use Line

Reply:
this is method of `IList`. not Linq
https://docs.microsoft.com/en-us/dotnet/api/system.collections.ilist.contains